WordPress/Classes/Networking/AccountServiceRemoteREST.m

```diff
@@ -34,24 +34,18 @@ - (void)getBlogsWithSuccess:(void (^)(NSArray *))success failure:(void (^)(NSErr
           }];
 }
 
-- (void)getDetailsWithSuccess:(void (^)(NSDictionary *userDetails))success failure:(void (^)(NSError *error))failure
+- (void)getDetailsForAccount:(WPAccount *)account success:(void (^)(RemoteUser *remoteUser))success failure:(void (^)(NSError *error))failure
 {
     NSString *path = @"me";
     [self.api GET:path
        parameters:nil
           success:^(AFHTTPRequestOperation *operation, id responseObject) {
-              if (success) {
-                  NSString *email = responseObject[@"email"];
-                  NSNumber *primaryBlogId = responseObject[@"primary_blog"];
-                  NSMutableDictionary *userDetails = [NSMutableDictionary new];
-                  if (email) {
-                      userDetails[@"email"] = email;
-                  }
-                  if (primaryBlogId) {
-                      userDetails[@"primary_blog"] = primaryBlogId;
-                  }
-                  success(userDetails);
+              if (!success) {
+                  return;
               }
+              NSDictionary *response = (NSDictionary *)responseObject;
```

Comment:
Why not do `success:^(AFHTTPRequestOperation *operation, NSDictionary* responseObject) {` in the line above?

It'd only save us one line (removing the cast below) but the functionality would remain exactly the same and it would be a nice code shortener.